components/script/dom/domparser.rs

```diff
@@ -77,7 +77,6 @@ impl DOMParserMethods for DOMParser {
                 Ok(document)
             }
             Text_xml | Application_xml | Application_xhtml_xml => {
-                // FIXME: this should probably be FromParser when we actually parse the string (#3756).
```

Comment:
This probably has some behavior change, and thus needs a test.

In particular, this determines the initial `ReadyState` and such, so we should be able to test whether parsing a document with a subresource claims to be loaded or not from the start.

Though @jdm or @nox probably know this code better.

Reply:
please let me know how to do to start to test. :)

Reply:
https://github.com/servo/servo/blob/master/tests/wpt/README.md#writing-new-tests has instructions for creating new tests. For this one in particular, we want to test that the document returned from `(new DOMParser()).parseFromString("<html></html>", "text/xml")` has `readyState` equal to `complete`.